src/inference/src/ie_network_reader.cpp

```diff
@@ -361,10 +362,17 @@ CNNNetwork convert_to_cnnnetwork(std::shared_ptr<ngraph::Function>& function,
                 if (it_type != rtInfo.end()) {
                     const auto old_api_map_attr = std::dynamic_pointer_cast<ov::OldApiMapElementType>(it_type->second);
                     OPENVINO_ASSERT(old_api_map_attr != nullptr, "Failed to cast to ov::OldApiMapElementType");
-                    const auto type = old_api_map_attr->get();
-                    pre_input.tensor().set_element_type(type);
+                    const auto old_api_map_type = old_api_map_attr->get();
+                    const auto param_type = parameter->get_element_type();
 
-                    OPENVINO_ASSERT(!type.is_dynamic(), "Old API map does not support dynamic type");
+                    if ((param_type == ngraph::element::u8 && old_api_map_type.is_real())) {
+                        parameter->set_element_type(old_api_map_type);
+                        need_validate_nodes_and_infer_types = true;
+                    } else {
+                        pre_input.tensor().set_element_type(old_api_map_type);
+                    }
```

Comment:
Please explain why in one case we set type directly to the Parameter but in another but set tensor type which will be reinferred after the first type propagation?

Reply:
pre_input is i-th input to PrePostProcessor, so it adds Convert node to selected type instead of just changing tensor type. As some plugins do not support uint8 type, Convert(uint8) leads to errors, so for this case type is set directly to Parameter instead of inserting Convert.

Reply:
Please add this comment to the code.

Reply:
Done.